src/hotspot/share/gc/shenandoah/shenandoahMarkingContext.hpp

```diff
@@ -61,16 +61,15 @@ class ShenandoahMarkingContext : public CHeapObj<mtGC> {
   inline bool mark_weak(oop obj);
 
   // Simple versions of marking accessors, to be used outside of marking (e.g. no possible concurrent updates)
-  // TODO: Do these really need to be const?
-  inline bool is_marked(const oop) const;
-  inline bool is_marked_strong(const oop obj) const;
-  inline bool is_marked_weak(const oop obj) const;
-  inline bool is_marked_or_old(const oop obj) const;
-  inline bool is_marked_strong_or_old(const oop obj) const;
+  inline bool is_marked(oop) const;
+  inline bool is_marked_strong(oop obj) const;
+  inline bool is_marked_weak(oop obj) const;
+  inline bool is_marked_or_old(oop obj) const;
+  inline bool is_marked_strong_or_old(oop obj) const;
 
   inline HeapWord* get_next_marked_addr(const HeapWord* addr, const HeapWord* limit) const;
 
-  inline bool allocated_after_mark_start(const oop obj) const;
+  inline bool allocated_after_mark_start(oop obj) const;
   inline bool allocated_after_mark_start(const HeapWord* addr) const;
```

Comment:
Agree with Ramki here, unless there's a reason we prefer non-const args.

Reply:
I believe it is confusing because `oop` is a `typedef` for a pointer `oopDesc*`. So, the `const` here applies to the typedef and not to the underlying type (that is, we could still call non-const methods on `oop` inside the function, but we cannot change what `oop` points to). Also, these `const` modifiers are not present upstream.

Reply:
ok.

Reply:
Good point, so what one wanted here was `oop const obj` ... :-) But that might cause other changes, and not a big deal here, since it's fewer diffs wrt upstream, as you state.